util/work_queue.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ */
+#pragma once
+
+#include <atomic>
+#include <cassert>
+#include <cstddef>
+#include <condition_variable>
+#include <cstddef>
+#include <functional>
+#include <mutex>
+#include <queue>
+
+namespace rocksdb {
+
+/// Unbounded thread-safe work queue.
```

Comment:
Added a unit test for work_queue.h, since there is an out-of-box unit test from [facebook/zstd](https://github.com/facebook/zstd/blob/dev/contrib/pzstd/utils/test/WorkQueueTest.cpp). Currently all methods in the class are used except waitUntilFinished, so maybe we could keep the class `WorkQueue` as it was in zstd repo.